### PR TITLE
Splits Malf AI and traitor AI into two antag types, now has its own mode

### PR DESCRIPTION
**Pull request by Amelia0010**
### Intent of your Pull Request

spiritual revert of https://github.com/tgstation/tgstation/pull/13759, although it's not a 1:1

Basically:
Traitor AIs, be it in traitor rounds or IAA rounds, will no longer be able to use malf abilities.
Added a malf-only gamemode, that makes malf AIs

#### Changelog

:cl:  
rscdel: Traitor AIs can no longer use malf abilities
rscadd: Malf AI now has its own gamemode
/:cl:

### CODE REVIEW

**Comment by Lynameep** [comment]
Atmospherics is still the only thing an AI needs to completely destroy the station so this change won't really do much but further widen the gap between players who know atmos and the ones that dont.

**Comment by CMOisLing** [comment]
Should tator AI be under the same murderbone rules regular traitor operate under?

**Comment by Ratvar** [comment]
I don't think traitor AI should be grief-all: "I can plasmaflood roundstart" possible in every single gamemode but nukes is not a fun game, I would say, no matter how easy it is to EMP AI after it ruined the round for everyone

**Comment by AsV9** [comment]
Traitor-AI will be subject to the murderbone as they were before traitor and malf-AI became the same gamemode.

File: config/game_options.txt [hunk 1]
```diff
@@ -155,6 +157,7 @@ MIDROUND_ANTAG CLOCKWORK_CULT
 MIDROUND_ANTAG CHANGELING
 MIDROUND_ANTAG WIZARD
 #MIDROUND_ANTAG  MONKEY
+MIDROUND_ANTAG MALF
```
Comment (alexkar598) [inline config/game_options.txt:160]: Won't this take a random Alice person and turn them into an AI mid round?

Comment (Amelia0010) [reply config/game_options.txt:160]: no, midround_antag is whether or not the round-type will make new antags after the main ones die. You're thinking of `reroll_friendly` which is a var on the gamemode itself.

**Comment by Alek2ander** [comment]
Malf AI is too damn weak to be its own gamemode

**Comment by Bibby0110** [comment]
If you buff Malf AI a bit I see no problem with this.

**Comment by boodaliboo** [comment]
yukjyytrgewoiyh6gfvrdcscjkdcmkdnkdsfnkdfnfgnnj yes

**Comment by alexkar598** [comment]
@boodaliboo when you don't have anything to say other than "I like" or "I no like" just use thumbs up/thumbs down emojis instead of clogging comments

**Pull request merged**